geomesa-gt/geomesa-gt-partitioning/src/main/scala/org/locationtech/geomesa/gt/partition/postgis/dialect/package.scala

```diff
@@ -0,0 +1,426 @@
+        if (rollback) {
+          Try(cx.rollback())
+        } else {
+          cx.commit()
+        }
+      }
+      statement.close()
+    }
+  }
+
+  /**
+   * All the info for a feature type used by the partitioning scheme
+   *
+   * @param schema database schema (e.g. "public")
+   * @param name feature type name
+   * @param tables table names
+   * @param cols columns
+   * @param partitions partition config
+   */
+  case class TypeInfo(schema: String, name: String, tables: Tables, cols: Columns, partitions: PartitionInfo)
+
+  object TypeInfo {
+    def apply(schema: String, sft: SimpleFeatureType): TypeInfo = {
+      val tables = Tables(sft, schema)
+      val columns = Columns(sft)
+      val partitions = PartitionInfo(sft)
+      TypeInfo(schema, sft.getTypeName, tables, columns, partitions)
+    }
+  }
+
+  /**
+   * Tables used by the partitioning system
+   *
+   * @param schema database schema name (e.g. "public")
+   * @param view primary view of all partitions
+   * @param writeAhead write ahead table
+   * @param writeAheadPartitions recent partitions table
+   * @param mainPartitions main partitions table
+   * @param analyzeQueue analyze queue table
+   * @param sortQueue sort queue table
+   */
+  case class Tables(
+      schema: String,
+      view: TableConfig,
+      writeAhead: TableConfig,
+      writeAheadPartitions: TableConfig,
+      mainPartitions: TableConfig,
+      analyzeQueue: TableConfig,
+      sortQueue: TableConfig)
+
+  object Tables {
+
+    import Config.ConfigConversions
+
+    def apply(sft: SimpleFeatureType, schema: String): Tables = {
+      val view = TableConfig(schema, sft.getTypeName, None)
+      // we disable autovacuum for write ahead tables, as they are transient and get dropped fairly quickly
+      val writeAhead = TableConfig(schema, view.name.raw + WriteAheadTableSuffix, sft.getWriteAheadTableSpace, vacuum = false)
+      val writeAheadPartitions = TableConfig(schema, view.name.raw + PartitionedWriteAheadTableSuffix, sft.getWriteAheadPartitionsTableSpace, vacuum = false)
+      val mainPartitions = TableConfig(schema, view.name.raw + PartitionedTableSuffix, sft.getMainTableSpace)
+      val analyzeQueue = TableConfig(schema, view.name.raw + AnalyzeTableSuffix, sft.getMainTableSpace)
+      val sortQueue = TableConfig(schema, view.name.raw + SortTableSuffix, sft.getMainTableSpace)
+      Tables(schema, view, writeAhead, writeAheadPartitions, mainPartitions, analyzeQueue, sortQueue)
+    }
+  }
+
+  /**
+   * Table configuration
+   *
+   * @param name table name
+   * @param tablespace table space
+   * @param storage storage opts (auto vacuum)
+   */
+  case class TableConfig(name: TableName, tablespace: TableSpace, storage: String)
+
+  object TableConfig {
+    def apply(schema: String, name: String, tablespace: Option[String], vacuum: Boolean = true): TableConfig = {
+      val storage = if (vacuum) { "" } else { " WITH (autovacuum_enabled = false)" }
```

Comment:
Hmm.... I wonder if one can turn off autovacuum globally.  If so, should this respect that or be trying to override "bad" system settings?  Either way, it may be worth explaining which tables have autovacuum set and which do not in the docs.

Reply:
you can disable auto vacuum at a global level, in which case this setting wouldn't affect anything. in that case, you're responsible to managing vacuums on your own. it's not inherently worse, just different. i disabled it on the write ahead tables because they're short-lived, so vacuuming them is needless overhead.